### PR TITLE
Contacts component with feedback form created + minor changes

### PR DESCRIPTION
**Pull request by ziviledauderiene**
screenshot022

- New feature: Feedback form with validations on every keypress, and onBlur. Submit button is enabled only when the whole form is valid.
- Small bug fix in Title comp.
- Minor styling changes in Card comp.

### CODE REVIEW

File: src/components/Contacts/Contacts.tsx [hunk 1]
```diff
@@ -1,7 +1,230 @@
...
+
+  checkFormValidity() {
+    this.setState({
+      formIsValid:
+        this.nameIsValid && this.emailIsValid && this.messageIsValid && this.numberIsValid
+    });
+  }
+
+  updateInputValidations() {
+    this.nameIsValid = (this.state.name as string) != '';
```
Comment (JuliaShavyrova) [inline src/components/Contacts/Contacts.tsx]: have you tried `!!this.state.name`?

File: src/components/Contacts/Contacts.tsx [hunk 2]
```diff
@@ -1,7 +1,230 @@
...
+    this.emailIsValid = emailRegEx.test(this.state.email as string);
```
Comment (JuliaShavyrova) [inline src/components/Contacts/Contacts.tsx]: why do you need `as` here?

File: src/components/Contacts/Contacts.tsx [hunk 3]
```diff
@@ -1,7 +1,230 @@
...
+    this.numberIsValid =
+      phoneRegEx.test(this.state.number as string) ||
+      (this.state.number as string) === '' ||
+      this.state.number === undefined;
+  }
+
+  formSubmitHandler(event: React.MouseEvent<HTMLButtonElement>) {
+    event.preventDefault();
+    const feedback: Feedback = {
+      id: Math.random(),
```
Comment (JuliaShavyrova) [inline src/components/Contacts/Contacts.tsx]: no need to add id

File: src/components/Contacts/Contacts.tsx [hunk 4]
```diff
@@ -1,7 +1,230 @@
...
+                <p className={styles.label}>
+                  Do you want to get the latest news and updates about our service?{' '}
+                  <span className={styles.required}>*</span>
+                </p>
+                <div className={styles.radioContainer}>
+                  <RadioInput
+                    name="subscribe"
+                    value="yes"
+                    changeHandler={this.inputChangeHandler.bind(this)}
+                    checked={this.state.subscribe as boolean}
```
Comment (JuliaShavyrova) [inline src/components/Contacts/Contacts.tsx]: please avoid using `as` when it's possible as far as it forces a compiler to think about variable as defined type and miss real type check that should be (in your case typescript should know that this.state.subscribe is a boolean according to `this.state` type). 
and also `this.state.subscribe as boolean` can be replaced with `!!this.state.subscribe`

File: src/components/Contacts/Contacts.tsx [hunk 5]
```diff
@@ -1,7 +1,230 @@
...
+
+class Contacts extends React.Component<ContactsProps, ContactsState> {
+  thankYouCard: React.RefObject<HTMLDivElement>;
+  nameIsValid: boolean = false;
+  emailIsValid: boolean = false;
+  messageIsValid: boolean = false;
+  numberIsValid: boolean = false;
+  constructor(props: ContactsProps) {
+    super(props);
+    this.state = {
```
Comment (JuliaShavyrova) [inline src/components/Contacts/Contacts.tsx:41]: missed?
https://react-typescript-cheatsheet.netlify.app/docs/basic/getting-started/class_components/
```suggestion
    this.state: ContactsState = {
```

File: src/components/Contacts/Contacts.tsx [hunk 6]
```diff
@@ -1,7 +1,230 @@
...
+                inputValue={this.state.name as string}
+                required={true}
+                valid={this.nameIsValid}
+                touched={this.state.nameTouched}
+                errorMessage="Name cannot be empty."
+                blurHandler={this.inputBlurHandler.bind(this)}
+              />
+              <Input
+                valueName="email"
+                changeHandler={this.inputChangeHandler.bind(this)}
```
Comment (JuliaShavyrova) [inline src/components/Contacts/Contacts.tsx]: please bind handlers once in constructor

File: src/components/Contacts/Input/Input.tsx [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+  capitalizeInputName() {
+    return this.props.valueName.slice(0, 1).toUpperCase() + this.props.valueName.slice(1);
+  }
+
+  render() {
+    return (
+      <>
+        <div className={styles.inputContainer}>
+          <label className={styles.label} htmlFor={`${this.props.valueName}-input`}>
+            {`${this.capitalizeInputName()}`}{' '}
```
Comment (JuliaShavyrova) [inline src/components/Contacts/Input/Input.tsx]: if you need space here, move it inside template string

File: src/components/Contacts/Input/Input.tsx [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+            {this.props.required && <span className={styles.required}>*</span>}
+          </label>
+          <input
+            type={this.props.type}
+            className={styles.input}
+            placeholder={this.props.placeholder || `Enter your ${this.capitalizeInputName()}`}
+            id={`${this.props.valueName}-input`}
+            value={this.props.inputValue}
+            onChange={this.props.changeHandler.bind(this)}
```
Comment (JuliaShavyrova) [inline src/components/Contacts/Input/Input.tsx]: why do you need bind here?

File: src/components/Contacts/RadioInput/RadioInput.tsx [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+        <label className={styles.radioLabel} htmlFor={`${this.props.value}-radio`}>
+          {this.capitalizeValue()}
+        </label>
+        <input
+          className={styles.radioInput}
+          type="radio"
+          name={this.props.name}
+          value={this.props.value}
+          id={`${this.props.value}-radio`}
+          onChange={this.props.changeHandler.bind(this)}
```
Comment (JuliaShavyrova) [inline src/components/Contacts/RadioInput/RadioInput.tsx]: the same question about bind :)

File: src/components/Header/Header.tsx [hunk 10]
```diff
@@ -55,4 +55,4 @@ const Header: React.FunctionComponent = () => {
   );
 };
 
-export default Header;
+export default React.memo(Header); // can we wrap like this? Is it worth?
```
Comment (JuliaShavyrova) [inline src/components/Header/Header.tsx]: you can :)

File: src/components/UI/Card/Card.tsx [hunk 11]
```diff
@@ -2,7 +2,11 @@ import React, { useContext } from 'react';
 import { ThemeContext, ThemeContextConfig, Theme } from '../../../store/theme-context';
 import styles from './Card.module.scss';
 
-const Card: React.FunctionComponent = ({ children }) => {
+interface CardProps {
+  children?: React.ReactNode;
+}
```
Comment (JuliaShavyrova) [inline src/components/UI/Card/Card.tsx]: no need to create extra interface, `FunctionComponent` definition already includes `children?: ReactNode | undefined`

File: src/store/theme-context.tsx [hunk 12]
```diff
@@ -20,7 +20,9 @@ export const ThemeContext = React.createContext<ThemeContextConfig>({
   toggleTheme: () => {}
 });
 
-export const ThemeProvider: React.FunctionComponent = ({ children }) => {
+export const ThemeProvider: React.FunctionComponent<{ children?: React.ReactNode }> = ({
```
Comment (JuliaShavyrova) [inline src/store/theme-context.tsx]: the same about children

File: src/components/Contacts/Contacts.tsx (hunk 5 above)
Comment (ziviledauderiene) [reply src/components/Contacts/Contacts.tsx:41]: IDE doesn't allow to add it here. Only in 32nd line.
screenshot023

File: src/components/Contacts/Input/Input.tsx (hunk 7 above)
Comment (ziviledauderiene) [reply src/components/Contacts/Input/Input.tsx]: Of course, changed it. But it was automatically added when formatting. Is there anything wrong with these? I noticed that formatter adds them quite often.

File: src/components/Contacts/Contacts.tsx (hunk 5 above)
Comment (JuliaShavyrova) [reply src/components/Contacts/Contacts.tsx:41]: seems like i shared an incorrect link :) it's fine, just leave your initial code version

File: src/components/Contacts/Input/Input.tsx (hunk 7 above)
Comment (JuliaShavyrova) [reply src/components/Contacts/Input/Input.tsx]: it just looks messy

File: src/components/Contacts/Contacts.tsx [hunk 13]
```diff
@@ -1,7 +1,228 @@
...
+
+  checkFormValidity() {
+    this.setState({
+      formIsValid:
+        this.nameIsValid && this.emailIsValid && this.messageIsValid && this.numberIsValid
+    });
+  }
+
+  updateInputValidations() {
+    this.nameIsValid = this.state.name != '';
```
Comment (JuliaShavyrova) [inline src/components/Contacts/Contacts.tsx]: please use strict equality according to best practices

File: src/components/Contacts/Contacts.tsx [hunk 14]
```diff
@@ -1,7 +1,228 @@
...
+    this.emailIsValid = emailRegEx.test(this.state.email || '');
+    this.messageIsValid = this.state.message != '';
+    this.numberIsValid =
+      phoneRegEx.test(this.state.number || '') || this.state.number === undefined;
```
Comment (JuliaShavyrova) [inline src/components/Contacts/Contacts.tsx]: why not like this?
`!this.state.number || phoneRegEx.test(this.state.number)`

File: src/components/Contacts/Contacts.tsx (hunk 6 above)
Comment (JuliaShavyrova) [reply src/components/Contacts/Contacts.tsx]: still see multiple bind :)

**Review by JuliaShavyrova — APPROVED** [review]

**Pull request merged**